examples/psyir/modify.py

```diff
@@ -0,0 +1,66 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: S. Siso, STFC Daresbury Lab
+
+'''A simple Python script showing how to modify a PSyIR tree. In order to use
+it you must first install PSyclone. See README.md in the top-level psyclone
+directory.
+
+Once you have psyclone installed, this script may be run by doing:
+
+>>> python modify.py
+
+This will first create a tree as specified in the create.py file and
```

Comment:
I completely agree about re-using create. You'll probably not believe me but I was actually going to suggest that in the last review :-). However, I need convincing that just importing it wholesale is the right way to do it. I would naturally expect that the code in `create.py` would be turned into a method (which is called from a `__main__`) and the method would be imported and called from `modify.py`. Did you do it this way as the variables are naturally accessible for modification, or was it more to do with minimal code changes, or some other reason? If it was the first reason then wouldn't it be better to control which variables are accessible externally i.e. only make SYMBOL_TABLE, CONTAINER and TMP_SYMBOL importable?

Reply:
I did it that way to maintain the create.py untouched, but since you mentioned it, I prefer having a main section an a (importable) function which looks much better than all in the global scope imo. (I think I already proposed this style change before but was not done :))

Reply:
You did, I remember :-)